packages/core/state/interaction/logics.ts

```diff
@@ -499,6 +501,105 @@ const openWithLogic = createLogic({
 });
 
+/**
+ * Interceptor responsible for translating an EDIT_FILES action into a progress tracked
+ * series of edits on the files currently selected.
+ */
+const editFilesLogic = createLogic({
+    async process(deps: ReduxLogicDeps, dispatch, done) {
+        const fileService = interactionSelectors.getFileService(deps.getState());
+        const fileSelection = selection.selectors.getFileSelection(deps.getState());
+        const sortColumn = selection.selectors.getSortColumn(deps.getState());
+        const annotationNameToAnnotationMap = metadata.selectors.getAnnotationNameToAnnotationMap(
+            deps.getState()
+        );
+        const {
+            payload: { annotations, filters },
+        } = deps.action as EditFiles;
+
+        // Gather up the files for the files selected currently
+        // if filters is present then actual "selected" files
+        // are the ones that match the filters, this happens when
+        // editing a whole folder for example
+        let filesSelected;
+        if (filters) {
+            const fileSet = new FileSet({
+                filters,
+                fileService,
+                sort: sortColumn,
+            });
+            const totalFileCount = await fileSet.fetchTotalCount();
+            filesSelected = await fileSet.fetchFileRange(0, totalFileCount);
+        } else {
+            filesSelected = await fileSelection.fetchAllDetails();
+        }
+
+        // Break files into batches of 10 File IDs
+        const fileIds = filesSelected.map((file) => file.uid);
+        const batches = chunk(fileIds, 10);
+
+        // Dispatch an event to alert the user of the start of the process
+        const editRequestId = uniqueId();
+        const editProcessMsg = "Editing files in progress.";
+        // TODO: Lyndsay's design did not include a progress bar for editing files
+        // nor a way to cancel the process. This is a placeholder for now.
+        dispatch(processStart(editRequestId, editProcessMsg, noop));
+
+        // Track the total number of files edited
+        let totalFileEdited = 0;
+
+        // Throttled progress dispatcher
+        const onProgress = throttle(() => {
+            dispatch(
+                processProgress(
+                    editRequestId,
+                    totalFileEdited / fileIds.length,
+                    editProcessMsg,
+                    noop
+                )
+            );
+        }, 1000);
+
+        try {
+            // Begin editing files in batches
+            for (const batch of batches) {
+                // Asynchronously begin the edit for each file in the batch
+                const promises = batch.map(
+                    (fileId) =>
+                        new Promise<void>(async (resolve, reject) => {
+                            try {
+                                await fileService.editFile(
+                                    fileId,
+                                    annotations,
```

Comment:
I see... I think the request to get MMS "editable data" was to for insertion right here where this `annotations` parameter is. However, that is a very slow `GET` fetch so I don't think we should necessarily switch to that. Instead, can we just pass here the annotation we want to replace the values for? Lets check up on the way the `PATCH` method works we might be golden with just switching to that.

Reply:
Alright found the logic in MMS, this is the logical path:
https://github.com/AllenCellSoftware/metadata-management-service/blob/master/src/main/java/org/alleninstitute/aics/metadatamanagementservice/filemetadata/FileMetadataController.java#L92
https://github.com/AllenCellSoftware/metadata-management-service/blob/master/src/main/java/org/alleninstitute/aics/metadatamanagementservice/filemetadata/FileIngestionService.java#L101C13-L101C40
https://github.com/AllenCellSoftware/metadata-management-service/blob/master/src/main/java/org/alleninstitute/aics/metadatamanagementservice/filemetadata/FileIngestionService.java#L171-L205

So based on the last chunk of code there we actually want that boolean block of `replaceOldValues` to happen which does not happen with the `PATCH` and does happen with the `PUT`. Furthermore, it only replaces annotations that are included in the request with new values. So all to get this working that we need to do is for the `annotations` parameter in this BFF code assign `annotations` equal to just the annotation the user modified in the modal with the values they want to replace with from the modal. It seems like that may already be happening so everything might just be all good.

Reply:
Yes, currently we are only passing the annotations that we want to edit

Reply:
Just to be clear, this is currently working as is